{"error":"TypeError: (intermediate value).getTime is not a function\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:19:38)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:20:15.183Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:75:38\n    at <anonymous>","level":"error","message":"BadRequest","timestamp":"2018-05-20T17:20:15.214Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:20:15.221Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:20:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:22:59.244Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:22:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:23:48.971Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:75:38\n    at <anonymous>","level":"error","message":"BadRequest","timestamp":"2018-05-20T17:23:49.002Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:23:49.008Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:23:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:24:36.940Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:75:38\n    at <anonymous>","level":"error","message":"BadRequest","timestamp":"2018-05-20T17:24:36.967Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:24:36.972Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:23:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:27:14.518Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:23:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:27:23.668Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:23:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:28:20.869Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:23:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:28:35.925Z"}
{"error":"SyntaxError: Unexpected token ' in JSON at position 3\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\body-parser\\lib\\types\\json.js:64:17)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\body-parser\\lib\\read.js:96:18\n    at IncomingMessage.onEnd (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\raw-body\\index.js:135:7)\n    at Object.onceWrapper (events.js:313:30)\n    at emitNone (events.js:111:20)\n    at IncomingMessage.emit (events.js:208:7)\n    at endReadableNT (_stream_readable.js:1064:12)\n    at _combinedTickCallback (internal/process/next_tick.js:138:11)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:28:50.647Z"}
{"error":"SyntaxError: Unexpected token ' in JSON at position 3\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\body-parser\\lib\\types\\json.js:64:17)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\body-parser\\lib\\read.js:96:18\n    at IncomingMessage.onEnd (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\raw-body\\index.js:135:7)\n    at Object.onceWrapper (events.js:313:30)\n    at emitNone (events.js:111:20)\n    at IncomingMessage.emit (events.js:208:7)\n    at endReadableNT (_stream_readable.js:1064:12)\n    at _combinedTickCallback (internal/process/next_tick.js:138:11)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:28:57.188Z"}
{"error":"SyntaxError: Unexpected token d in JSON at position 3\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\body-parser\\lib\\types\\json.js:64:17)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\body-parser\\lib\\read.js:96:18\n    at IncomingMessage.onEnd (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\raw-body\\index.js:135:7)\n    at Object.onceWrapper (events.js:313:30)\n    at emitNone (events.js:111:20)\n    at IncomingMessage.emit (events.js:208:7)\n    at endReadableNT (_stream_readable.js:1064:12)\n    at _combinedTickCallback (internal/process/next_tick.js:138:11)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:29:08.757Z"}
{"error":"Error: invalid json\n    at parse (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\body-parser\\lib\\types\\json.js:60:15)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\body-parser\\lib\\read.js:96:18\n    at IncomingMessage.onEnd (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\raw-body\\index.js:135:7)\n    at Object.onceWrapper (events.js:313:30)\n    at emitNone (events.js:111:20)\n    at IncomingMessage.emit (events.js:208:7)\n    at endReadableNT (_stream_readable.js:1064:12)\n    at _combinedTickCallback (internal/process/next_tick.js:138:11)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:29:25.815Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:23:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:30:43.681Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:23:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:30:56.038Z"}
{"error":"TypeError: Data must be a string or a buffer\n    at Hash.update (crypto.js:99:16)\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:23:50)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:31:13.230Z"}
{"error":"TypeError: Date.now is not a function\n    at model.<anonymous> (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\db\\mongoose\\schemas\\UserSchema.js:18:26)\n    at callMiddlewareFunction (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:402:23)\n    at model.next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:58:7)\n    at _next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:106:10)\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\kareem\\index.js:427:38)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T17:34:51.001Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:69:38\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)","level":"error","message":"{ _id: 5b01b211f3762f3d30e12235,\n  first_name: 'sample',\n  last_name: 'sample',\n  password: '+sd6lyEzaV9H4+CkzfRjAXkI5Yk5MtLswCeYGHec7HE=',\n  email: 'sample@test.com',\n  created_at: 2018-05-20T17:36:17.989Z,\n  password_salt: '1526837777989',\n  __v: 0 }","timestamp":"2018-05-20T17:36:18.030Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:36:18.038Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:69:38\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)","level":"error","message":"{ _id: 5b01b211f3762f3d30e12235,\n  first_name: 'sample',\n  last_name: 'sample',\n  password: '+sd6lyEzaV9H4+CkzfRjAXkI5Yk5MtLswCeYGHec7HE=',\n  email: 'sample@test.com',\n  created_at: 2018-05-20T17:36:17.989Z,\n  password_salt: '1526837777989',\n  __v: 0 }","timestamp":"2018-05-20T17:40:59.635Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:40:59.643Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:69:38\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)","level":"error","message":"{ _id: 5b01b211f3762f3d30e12235,\n  first_name: 'sample',\n  last_name: 'sample',\n  password: '+sd6lyEzaV9H4+CkzfRjAXkI5Yk5MtLswCeYGHec7HE=',\n  email: 'sample@test.com',\n  created_at: 2018-05-20T17:36:17.989Z,\n  password_salt: '1526837777989',\n  __v: 0 }","timestamp":"2018-05-20T17:42:38.782Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:42:38.789Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:82:23\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"BadRequest","timestamp":"2018-05-20T17:45:58.088Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:124:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:45:58.095Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:84:23\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"BadRequest","timestamp":"2018-05-20T17:47:13.161Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:47:13.169Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:84:23\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"BadRequest","timestamp":"2018-05-20T17:49:06.342Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:49:06.350Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:126:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:52:19.449Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:89:54\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)","level":"error","message":"{ _id: 5b01b662d78b62312c8a71a4,\n  user_id: 5b01b5b4fba60c399c245b46,\n  created_at: 2018-05-20T17:54:42.318Z,\n  nonce: 9593752120,\n  expiry: 2018-05-20T17:54:42.000Z,\n  closed: false,\n  updated_at: 2018-05-20T17:54:42.319Z,\n  __v: 0 }","timestamp":"2018-05-20T17:54:43.358Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:124:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:54:43.367Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:134:28\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)","level":"error","message":"{ _id: 5b01b68c261f8008045a6821,\n  user_id: 5b01b5b4fba60c399c245b46,\n  created_at: 2018-05-20T17:55:24.179Z,\n  nonce: 6433753401,\n  expiry: 2018-05-20T17:55:24.000Z,\n  closed: false,\n  updated_at: 2018-05-20T17:55:24.179Z,\n  __v: 0 }","timestamp":"2018-05-20T17:55:25.263Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:82:23\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"BadRequest","timestamp":"2018-05-20T17:56:26.650Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:124:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:56:26.657Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at AuthenticationService.authenticate (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:124:9)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at Route.dispatch (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:112:3)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:277:22\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)\n    at Function.handle (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:176:3)\n    at router (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:46:12)\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:312:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:280:7\n    at Function.process_params (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:330:12)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\index.js:271:10)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T17:58:24.114Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:134:28\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)","level":"error","message":"{ _id: 5b01b7596b87e12c5ca9aec2,\n  user_id: 5b01b7596b87e12c5ca9aec1,\n  created_at: 2018-05-20T17:58:49.952Z,\n  nonce: 4638054617,\n  expiry: 2018-05-20T17:58:49.000Z,\n  closed: false,\n  updated_at: 2018-05-20T17:58:49.953Z,\n  __v: 0 }","timestamp":"2018-05-20T17:58:50.996Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:134:28\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)","level":"error","message":"{ _id: 5b01b7e23407fb216c927263,\n  user_id: 5b01b7e13407fb216c927262,\n  created_at: 2018-05-20T18:01:06.026Z,\n  nonce: 7783934617,\n  expiry: 2018-05-20T18:01:06.000Z,\n  closed: false,\n  updated_at: 2018-05-20T18:01:06.027Z,\n  __v: 0 }","timestamp":"2018-05-20T18:01:07.043Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:146:13\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T18:02:42.384Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:146:13\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T18:05:32.386Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:147:13\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T18:06:46.983Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:147:13\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T18:10:58.948Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:147:13\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T18:11:29.379Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:147:13\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T18:11:59.160Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:147:13\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T18:12:27.477Z"}
{"error":"    at new CustomError (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\misc\\Error.js:7:14)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:147:13\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"InvalidRequest","timestamp":"2018-05-20T18:14:14.457Z"}
{"error":"TypeError: User.delete is not a function\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\routes\\users.js:39:16\n    at Layer.handle [as handle_request] (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\express\\lib\\router\\route.js:131:13)\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\app\\services\\AuthenticationService.js:180:32\n    at C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\model.js:3949:16\n    at process.nextTick (C:\\Users\\sfast\\Documents\\IMT18\\project\\server\\node_modules\\mongoose\\lib\\query.js:2016:28)\n    at _combinedTickCallback (internal/process/next_tick.js:131:7)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error","message":"FatalError","timestamp":"2018-05-20T18:31:15.171Z"}
